### PR TITLE
Allow relationships with undefined properties

### PR DESCRIPTION
**Pull request by ndowmon**
### Changed

- Allow relationships to have `undefined` top-level properties

### CODE REVIEW

File: packages/integration-sdk-testing/src/jest.ts [hunk 1]
```diff
@@ -536,7 +536,24 @@ export function toMatchDirectRelationshipSchema<T extends ExplicitRelationship>(
     newRelationshipSchema.properties._type = { const: params._type };
   }
 
-  return toMatchSchema(received, newRelationshipSchema);
+  return toMatchSchema(
+    removeUndefinedProperties(received),
```
Comment (ndowmon) [inline packages/integration-sdk-testing/src/jest.ts:540]: AJV has some strange behavior in that it doesn't treat `undefined ` as `null`. 

Therefore, using the `directRelationshipSchema`, which enforces that all properties must be primitive (no arrays or objects) doesn't allow one to produce relationships with `undefined` properties:

```ts
      {
        additionalProperties: {
          anyOf: [
            { type: 'boolean' },
            { type: 'integer' },
            { type: 'null' },
            { type: 'number' },
            { type: 'string' },
          ],
        },
```

**Review by adam-in-ict — APPROVED** [review]

**Review by VDubber — APPROVED** [review]
Looks good!

File: packages/integration-sdk-testing/src/jest.ts [hunk 2]
```diff
@@ -536,7 +536,24 @@ export function toMatchDirectRelationshipSchema<T extends ExplicitRelationship>(
...
+
+function removeUndefinedProperties<T extends Entity | ExplicitRelationship>(
+  received: T | T[],
+) {
+  if (!Array.isArray(received)) {
+    received = [received];
+  }
+  return received.map((r) => {
+    for (const [key, value] of Object.entries(r)) {
+      if (value === undefined) delete r[key];
```
Comment (austinkelleher) [inline packages/integration-sdk-testing/src/jest.ts:553]: Should we be mutating the original object or perhaps we should create a new object and copy over _defined_ values?

**Review by austinkelleher — APPROVED** [review]
Looks good. Left a minor comment.

**Review by VDubber — APPROVED** [review]

**Pull request merged**